compiler/rustc_mir/src/transform/remove_unneeded_drops.rs

```diff
@@ -11,6 +11,10 @@ pub struct RemoveUnneededDrops;
 
 impl<'tcx> MirPass<'tcx> for RemoveUnneededDrops {
     fn run_pass(&self, tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
+        if !tcx.consider_optimizing(|| format!("RemoveUnneededDrops {:?} ", body.source.def_id())) {
```

Comment:
this should happen in the loop on `opt_finder.optimizations` or directly in `RemoveUnneededDropsOptimizationFinder